modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/construct/RouteComponentObjectFactory.java

```diff
@@ -72,25 +73,25 @@ public Object doGetObject() throws Exception {
     return withContextClassLoader(classLoader, () -> {
       initialiser.runOnce();
 
-      final MessageProcessorChain nestedChain;
+      final MessageProcessorChain chain;
 
       if (nestedProcessors != null) {
-        nestedChain = newChain(empty(), nestedProcessors);
+        chain = newChain(empty(), nestedProcessors);
         final StreamingManager streamingManager =
             ((MuleContextWithRegistry) muleContext).getRegistry().lookupObject(StreamingManager.class);
         model.getNestedComponents().stream()
             .filter(component -> component instanceof NestedChainModel)
             .findFirst()
-            .ifPresent(chain -> parameters.put(chain.getName(),
-                                               new ProcessorChainValueResolver(streamingManager, nestedChain)));
+            .ifPresent(chainModel -> parameters.put(chainModel.getName(),
+                                                    new ProcessorChainValueResolver(model, chain, streamingManager)));
```

Comment:
Shouldn't the first parameter passed to the ProcessorChainValueResolver here be the `chainModel` instead of `model`? I'm comparing it with [the other change for scopes](https://github.com/mulesoft/mule/pull/13695/files#diff-f68ed831a03cc038b795e4dad14a5ecd0aa4ff32e576fda99df665fef76f1613L83-R85).

Reply:
for scopes, the element defined through sdk-api is the Chain itself. For Routers, it's the Route. The codes are different because the use cases are